nova/scheduler/filters/cpu_info_migration_filter.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2021 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import six
+
+from oslo_log import log as logging
+from oslo_serialization import jsonutils
+
+from nova.context import get_admin_context
+from nova.exception import ComputeHostNotFound
+from nova.objects.compute_node import ComputeNode
+from nova.scheduler import filters
+from nova.scheduler.utils import request_is_rebuild
+from nova.scheduler.utils import request_is_resize
+
+LOG = logging.getLogger(__name__)
+
+
+class CpuInfoMigrationFilter(filters.BaseHostFilter):
+    """HostFilter checking the source flags are a subset of the target host."""
+
+    # Instance type and host capabilities do not change within a request
+    run_filter_once_per_request = True
+
+    RUN_ON_REBUILD = False
+
+    def __init__(self):
+        super(CpuInfoMigrationFilter, self).__init__()
+        self._source_cpu_flags = None
+
+    def filter_all(self, filter_obj_list, spec_obj):
+        self._source_cpu_flags = None  # Reset in case of re-use of filter
```

Comment:
True, with the thread safety-issue. I think, I have a simpler solution though:
Instead of calling `super().filter_all(...)`, I just do 
```python
    def filter_all(self, filter_obj_list, spec_obj):
        [...]
        for host_state in filter_obj_list:
            if self.host_passes(host_state, source_cpu_flags):
                yield host_state

    def host_passes(self, host_state, source_cpu_flags):
```
And take the 
That way the filter is much more self-contained.

Reply:
Well, I use a different function name than ´host_passes` too, to avoid confusion.